### ISSUE TITLE
Section on OpenPGP Based key data is too restrictive, hampers compatibility and encourages insecure practices

### ISSUE THREAD

**Issue by teythoon** [autocrypt/autocrypt#436]
https://autocrypt.org/level1.html#openpgp-based-key-data says:

> The keydata sent by an Autocrypt-enabled Level 1 MUA MUST consist of an OpenPGP “Transferable Public Key” containing exactly these five OpenPGP packets:
> 
>         a signing-capable primary key
>         a user id
>         a self signature over the user id by the primary key
>         an encryption-capable subkey
>         a binding signature over the subkey by the primary key

I see multiple problems with this, all stemming from the fact that the spec assumes that the primary key is signing capable.

  - This precludes certificates typically used with smart cards
  - If the primary key is not signing capable, then the recipient cannot check signatures made by the certificate holder
  - Signing capable primary keys are strictly less secure, because it opens an attack vector where signatures over data are reused as certification signatures, a variant of the Shambles attack

Furthermore, restricting to one encryption-capable subkey means that Autocrypt cannot be used to communicate encryption subkey updates in advance.  Being able to do that would be neat for rapid encryption subkey updates that approximate forward secrecy.

I've spoken privately with @Valodim about this issue, and he agrees it is an issue, but said (iirc) that it is not so pressing because no Autocrypt client actually enforces this.  However, I've discovered that this section of the spec has informed RNP's API: https://github.com/rnpgp/rnp/blob/master/include/rnp/rnp.h#L870

**Comment by teythoon** [comment]
Can you please shed some light on how this issue has been fixed, or failing that re-open the issue?

**Comment by hpk42** [comment]
On Mon, Jun 10, 2024 at 03:43 -0700, Justus Winter wrote:
> Can you please shed some light on how this issue has been fixed, or failing that re-open the issue?

see autocrypt announcement https://lists.mayfirst.org/pipermail/autocrypt/2024-June/000590.html

**Comment by teythoon** [comment]
This issue has been independently identified as important in the Autocrypt session at the OpenPGP Email Summit 2024.  But, I trust that someone will remember the issue without a written note, and will refrain from mentioning it again.

**Comment by Valodim** [comment]
as @teythoon mentioned above, I agree that this is an issue that should be addressed and thus reopened :)

**Comment by hpk42** [comment]
On Tue, Jun 11, 2024 at 00:43 -0700, Vincent Breitmoser wrote:
> as @teythoon mentioned above, I agree that this is an issue that should be addressed and thus reopened :)

As of now, there is no active maintainer group. 
Virtually all closed issues can be re-visited and taken care of
but there is no point in having years-old issues without any interaction. 
This was discussed at the OpenPGP summit and we agreed to, for the time being,
only allow issues/PRs if there are at least two parties caring. 

In other words: if you two, or someone else in the orbit, commit to, 
and assign yourself, to care for resolving this issue,
then i'd be fine to re-open it.

**Comment by dkg** [comment]
I'm interested in reopening it too.  The reason for the narrow spec in Autocrypt originally was twofold:

- to ensure that an implementation didn't feel the need to import a sprawling OpenPGP implementation to adopt autocrypt (no need to implement El Gamal, for example, or even third-party certifications), and
- to ensure that there would be mutual compatibility across Autocrypt clients.

I think both of those goals are still legitimate goals, while also sharing @teythoon's concern about hampering ecosystem evolution.  What would it look like to solve both?

**Comment by dkg** [comment]
One thing that might be concretely useful is to document a next-gen Autocrypt certificate profile that could be used as an argument to `sop generate-key --profile` .

**Comment by dkg** [comment]
It would also be useful to clarify whether we're looking only at certificate structure, or at pubkey algorithm selection as well.

**Comment by dkg** [comment]
I don't know who has permissions to assign themselves to this issue, but if either @teythoon or @Valodim (who have expressed an interest) are interested in having this assigned to them can't do the assignment, just let me know and i'll mark you as assigned.
